### PR TITLE
Added .codeclimate.yml

### PR DESCRIPTION
**Pull request by proton**
Issue https://github.com/sendgrid/sendgrid-ruby/issues/247

- added .codeclimate.yml
- added robocop to development dependencies

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=248) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=248) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-ruby?pullRequest=248) it.</sub>

File: .codeclimate.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    enabled: true
+  rubocop:
+    enabled: true
+ratings:
+  paths:
+  - "**.rb"
+exclude_paths:
+- examples/
+- spec/
+- test/
```
Comment (rahul26goyal) [inline .codeclimate.yml]: should we include gemfiles/ as well?

File: .codeclimate.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+engines:
+  duplication:
+    enabled: true
+    config:
+      languages:
+      - ruby
+  fixme:
+    enabled: true
+  rubocop:
+    enabled: true
```
Comment (rahul26goyal) [inline .codeclimate.yml]: we can add 
bundler-audit:
    enabled: true

**Comment by thinkingserious** [comment]
@proton

We have not been able to merge your Pull Request, but because you are awesome - we wanted to make sure you could still get a SendGrid Hacktoberfest shirt.  

Please go **fill [out our swag form](https://dx.sendgrid.com/swag) before Nov 5th** and we will send the shirt! (We know that you might have tried this before and it didn’t work, sorry about that!)

### **You have till Nov 5th to fill out this form** in order to get the Hacktoberfest shirt!

Thank you for contributing during Hacktoberfest! We hope to see you in the repos soon! Just so you know, we always give away a SendGrid shirt for your first ever non-Hacktoberfest PR that gets merged.

**Comment by thinkingserious** [comment]
Hello @proton,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Comment by mbernier** [comment]
@proton  & @rahul26goyal - 

If you are interested, we would love the opportunity to talk to you either/both of you about Hacktoberfest and our API libraries.

Our agenda would be: Explore what you liked and is there anything we can do to improve? 

You can grab a time [on my calendar](https://calendly.com/matt-bernier/30min) that works for you and we can have a chat on Google Hangout or Skype. If you prefer, you can email me using my GitHub username at my company’s domain.

Thank you so much,

Matt Bernier - @mbernier - SendGrid Developer Experience Product Manager
Elmer Thomas - @thinkingserious - SendGrid Developer Experience Engineer
@SendGrid

**Pull request merged**